extensions/cache/src/main/java/org/radargun/stages/cache/test/CacheInvocations.java

```diff
@@ -599,4 +604,83 @@ public Operation txOperation() {
          return async ? ASYNC_TX : NATIVE_TX;
       }
    }
+
+   public static final class GetViaStream<K, V extends Number> implements Invocation<Integer> {
+      public static final Operation GET_NULL = StreamingOperations.GET.derive("Null");
+      public static final Operation TX = StreamingOperations.GET.derive("TX");
+      private final StreamingOperations.StreamingCache<K> cache;
+      private final K key;
+      private Integer value;
+      private byte[] buffer;
+
+      public GetViaStream(StreamingOperations.StreamingCache<K> cache, K key, byte[] buffer) {
+         this.cache = cache;
+         this.key = key;
+         this.buffer = buffer;
+      }
+
+      @Override
+      public Integer invoke() {
+         Integer result = 0;
+         try (InputStream in = cache.getViaStream(key)) {
+            int read = 0;
+            while ((read = in.read(buffer)) != -1)
+               result = result + read;
+            return value = result;
+         } catch (IOException e) {
+            throw new RuntimeException(e);
+         } catch (NullPointerException e) {
```

Comment:
I think that `if (in == null) return null` is a slightly better way to deal with this...